tests/post-checkout.bats

```diff
@@ -2,62 +2,97 @@
 
 # load '/usr/local/lib/bats/load.bash'
 load "${BATS_PATH:?}/load.bash"
+source "/plugin/lib/common.sh"
+
+# Create fake "buildkite-agent"
+cp /plugin/tests/fake-buildkite-agent.sh /usr/local/bin/buildkite-agent
+chmod +x /usr/local/bin/buildkite-agent
 
 function unset_plugin_variables() {
-    unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_EXTERNAL_REPO
+    unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_REPO_URL
     unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER
-    unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_THIS_IS_FIRST_JOB
     unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION
-    unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION_FILE
+
+    buildkite-agent metadata unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_REPO_URL
+    buildkite-agent metadata unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_FOLDER
+    buildkite-agent metadata unset BUILDKITE_PLUGIN_EXTERNAL_BUILDKITE_VERSION
 }
 
 function setup() {
     unset_plugin_variables
 
     unset TEMPORARY_DIRECTORY
     TEMPORARY_DIRECTORY=$(mktemp -d)
+    pushd "${TEMPORARY_DIRECTORY}" >/dev/null
 }
```

Comment:
```suggestion
    rm -rf /tmp/metadata_store
}
```

Reply:
Yeah.... okay, I'll just remove the `buildkite-agent metadata unset` stuff and instead make a `buildkite-agent metadata clear-all`